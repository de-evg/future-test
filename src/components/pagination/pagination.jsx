import React, { useCallback } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { STEP } from "../../const";
import { ActionCreator } from "../../store/action";
import { NameSpace } from "../../store/reducers/root";

const START_PAGE = 1;

const Pagination = ({ userCount, currentStep, setStep }) => {
  const maxPage = Math.ceil(userCount / STEP);
  const currentPage = Math.ceil(currentStep / STEP);

  const handlePrevBtnClick = useCallback(() => {
    setStep(currentStep - STEP);
  }, [setStep, currentStep]);

  const handleNextBtnClick = useCallback(() => {
    setStep(currentStep + STEP);
  }, [setStep, currentStep]);

  const handleLastPageBtnClick = useCallback(() => {
    setStep(maxPage * STEP);
  }, [setStep, maxPage]);

  const handleFirstPageBtnClick = useCallback(() => {
    setStep(STEP);
  }, [setStep]);

  return (
    <div className="pagination">
      <button
        disabled={currentPage === START_PAGE}
        onClick={handlePrevBtnClick}
        className="pagination__arrow-left"
      ></button>
      <ul>
        <li className="pagination__firstPage">
          {currentPage !== START_PAGE && (
            <button onClick={handleFirstPageBtnClick} href="#">
              {START_PAGE}
            </button>
          )}
        </li>
        <li className="pagination__current">{currentPage}</li>
        <li className="pagination__lastPage">
          {currentPage !== maxPage && (
            <button onClick={handleLastPageBtnClick} href="#">
              {maxPage}
            </button>
          )}
        </li>
      </ul>
      <button
        disabled={currentPage === maxPage}
        onClick={handleNextBtnClick}
        className="pagination__arrow-right"
      ></button>
    </div>
  );
};

Pagination.propTypes = {
  userCount: PropTypes.number.isRequired,
  currentStep: PropTypes.number.isRequired,
  setStep: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  userCount: state[NameSpace.USERS_DATA].userCount,
  currentStep: state[NameSpace.USERS_DATA].currentStep,
});

const mapDispatchToProps = (dispatch) => ({
  setStep(newStep) {
    dispatch(ActionCreator.updateCurrentStep(newStep));
  },
});

export { Pagination };
export default connect(mapStateToProps, mapDispatchToProps)(Pagination);
