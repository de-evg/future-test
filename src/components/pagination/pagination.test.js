import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {Pagination} from './pagination';
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({adapter: new Adapter()});

it(`Render Pagination correctly`, () => {
    const wrapper = shallow(<Pagination userCount={500} currentStep={50} setStep={() => { }} />);
    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
});