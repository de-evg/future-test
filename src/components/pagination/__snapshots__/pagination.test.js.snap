// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Pagination correctly 1`] = `
<div
  className="pagination"
>
  <button
    className="pagination__arrow-left"
    disabled={true}
    onClick={[Function]}
  />
  <ul>
    <li
      className="pagination__firstPage"
    />
    <li
      className="pagination__current"
    >
      1
    </li>
    <li
      className="pagination__lastPage"
    >
      <button
        href="#"
        onClick={[Function]}
      >
        10
      </button>
    </li>
  </ul>
  <button
    className="pagination__arrow-right"
    disabled={false}
    onClick={[Function]}
  />
</div>
`;
